### PR TITLE
Stun Effects Refactor

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Refactors effects (stun, knockdown, knockout, daze, slow, superslow) into all being handled by apply_effects(), adjust_effects(), or set_effects()

Said procs now also send signals to source mob, meaning they can be manually cancelled without shitcode.

effect_flags can be used to check where the stuns come from, though currently they only show up in life.dm. (Example, some sort of ability that resists only explosive stuns)

TL;DR: No frontend changes. Made it significantly easier to cancel stuns in the future.

## Why It's Good For The Game

It isn't. This will add proc overhead to every stun in the game and every life check for every mob without even adding anything useful. Do NOT merge this!!!!!

## Changelog

:cl:
refactor: refactored how stuns/knockdowns/etc are handled in code to allow for manual cancellation without shitcode.
/:cl:

### CODE REVIEW

**Review by carlarctg — COMMENTED** [review]
they call me the web editor

**Comment by fira** [comment]
should just port tg effects :(

**Review by fira — APPROVED** [review]

**Review by harryob — APPROVED** [review]
> It isn't. This will add proc overhead to every stun in the game and every life check for every mob without even adding anything useful. Do NOT merge this!!!!!
you can't stop me contributor

File: code/datums/diseases/black_goo.dm [hunk 1]
```diff
@@ -146,10 +146,7 @@
 				target.AddDisease(new /datum/disease/black_goo)
 				to_chat(user, SPAN_XENOWARNING("<b>You sense your target is now infected.</b>"))
 
-	if(isSynth(target))
-		target.Slow(2)
-	else
-		target.Superslow(2) // Make them slower
+	target.apply_effect(2, SUPERSLOW)
```
Comment (Geevies) [inline code/datums/diseases/black_goo.dm:149]: this was specifically made to only regular slow synths

File: code/modules/mob/living/damage_procs.dm [hunk 2]
```diff
@@ -58,11 +58,25 @@
 /mob/living/proc/apply_internal_damage(var/damage = 0, var/organ)
 	return
 
+#define EFFECT_FLAG_LIFE (1>>0)
+#define EFFECT_FLAG_DEFAULT (1>>1)
```
Comment (Geevies) [inline code/modules/mob/living/damage_procs.dm:62]: pretty sure these bitflag arrows are pointing the wrong way

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/damage_procs.dm (hunk 2 above)
Comment (carlarctg) [reply code/modules/mob/living/damage_procs.dm:62]: durrrrrp

File: code/datums/diseases/black_goo.dm (hunk 1 above)
Comment (carlarctg) [reply code/datums/diseases/black_goo.dm:149]: but... typechecks... waaaaa

**Review by carlarctg — COMMENTED** [review]
Geves hates me (its true)
